Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs

```diff
@@ -38,15 +41,20 @@ public class CartItemsChangeEvent : UnityEvent<List<CartItem>> {}
         public UnityEvent OnPurhcaseCancelled = new UnityEvent();
         public PurchaseFailedEvent OnPurhchaseFailed = new PurchaseFailedEvent();
 
-        private Dictionary<string, ProductVariant> _idsToVariants = new Dictionary<string, ProductVariant>();
+        private struct ProductAndVariantPair {
```

Comment:
Once a pair is made, should it be mutable? If no, should we implement this with a constructor and getters? I think that would enforce the ability to make a pair, but not change it afterwards.